src/product_config_utils.rs

```diff
@@ -9,12 +9,28 @@ use tracing::{debug, error, warn};
 pub enum ConfigError {
     #[error("Invalid configuration found")]
     InvalidConfiguration,
+
+    #[error("Collected product config validation errors: {collected_errors:?}")]
+    ProductConfigErrors {
+        collected_errors: Vec<product_config::error::Error>,
+    },
 }
 
+/// This needs to be implemented by every generic (T) struct that may appear in [Role<T>]
+/// or the top level in order to determine where config properties are configured.
```

Comment:
```suggestion
/// This trait is used to compute configuration properties for products.
```

This is just a suggestion but the current version is also not really correct

Reply:
I would merge the existing comment with that one.